content/ngf/install/upgrade-version.md

```diff
@@ -0,0 +1,301 @@
+```
+
+If you installed NGINX Gateway from the experimental channel, use this instead:
+
+```shell
+kubectl kustomize "https://github.com/nginx/nginx-gateway-fabric/config/crd/gateway-api/experimental?ref=v{{< version-ngf >}}" | kubectl apply -f -
+```
+
+### Upgrade NGINX Gateway Fabric CRDs
+
+Run the following command to upgrade the CRDs:
+
+```shell
+kubectl apply -f https://raw.githubusercontent.com/nginx/nginx-gateway-fabric/v{{< version-ngf >}}/deploy/crds.yaml
+```
+
+{{< note>}} Ignore the following warning, as it is expected.
+
+```text
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply.
+```
+{{</ note >}}
+
+
+
+### Upgrade NGINX Gateway Fabric release
+
+{{< tabs name="upgrade-release" >}}
+
+{{% tab name="Helm" %}}
+
+{{< important >}} If you are using NGINX Plus and have a different Secret name than the default `nplus-license` name, specify the Secret name by setting `--set nginx.usage.secretName=<secret-name>` when running `helm upgrade`. {{< /important >}}
+
+To upgrade the release with Helm, you can use the OCI registry, or download the chart and upgrade from the source.
+
+If needed, replace `ngf` with your chosen release name.
+
+**Upgrade from the OCI registry**
+
+```shell
+helm upgrade ngf oci://ghcr.io/nginx/charts/nginx-gateway-fabric -n nginx-gateway
+```
+
+**Upgrade from sources**
+
+{{< include "/ngf/installation/helm/pulling-the-chart.md" >}}
+
+To upgrade, run the following command:
+
+```shell
+helm upgrade ngf . -n nginx-gateway
+```
+
+{{% /tab %}}
+
+{{% tab name="Manifests" %}}
+
+Select the deployment manifest that matches your current deployment from options available in the  [Deploy NGINX Gateway Fabric]({{< ref "/ngf/install/manifests.md#deploy-nginx-gateway-fabric-1">}}) section and apply it.
+
+{{% /tab %}}
+
+{{< /tabs>}}
+
+## Upgrade from v1.x to v2.x
+
+This section provides step-by-step instructions for upgrading NGINX Gateway Fabric from version 1.x to 2.x, highlighting key architectural changes, expected downtime, and important considerations for CRDs.
+
+To upgrade NGINX Gateway Fabric from version 1.x to the new architecture in version 2.x, you must uninstall the existing NGINX Gateway Fabric CRDs and deployment, and perform a fresh installation. This will cause brief downtime during the upgrade process.
+
+{{<note>}} You do not need to uninstall the Gateway API CRDs during the upgrade. These resources are compatible with the new NGINX Gateway Fabric version. {{</note>}}
+
+### Uninstall NGINX Gateway Fabric v1.x
+
+To remove the previous version 1.x of NGINX Gateway Fabric, follow these steps:
+
+First, run the following command to uninstall NGINX Gateway Fabric from the `nginx-gateway` namespace, and update `ngf` to your release name if it is different:
+
+```shell
+helm uninstall ngf -n nginx-gateway 
```

Comment:
Realizing it's a little weird that we have a general `helm uninstall` command here, but then for re-installing, we let users pick between helm or manifests.

Reply:
Yeah, I also realised when I was consolidating upgrade information there's instructions on going from OSS to Plus for Helm but not Manifests.

In this case, do Manifest users need to uninstall NGF from the namespace in a similar fashion?

Reply:
I'll be honest, I haven't tested how it works with the new architecture. It might not work as smoothly as it did before.